sklearn/cluster/tests/test_affinity_propagation.py

```diff
@@ -239,10 +254,10 @@ def test_affinity_propagation_random_state():
 
 
 @pytest.mark.parametrize("centers", [csr_matrix(np.zeros((1, 10))), np.zeros((1, 10))])
-def test_affinity_propagation_convergence_warning_dense_sparse(centers):
+def test_affinity_propagation_convergence_warning_dense_sparse(centers, global_dtype):
```

Comment:
This test should use assert_allclose

Reply:
Once again, if those are equal, can we keep `assert_array_equal`?